src/jit/morph.cpp

```diff
@@ -3464,7 +3473,7 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
                     structPassingKind howToPassStruct;
                     structBaseType = getArgTypeForStruct(objClass, &howToPassStruct, originalSize);
 
-#ifdef _TARGET_ARM64_
+#if defined(_TARGET_ARM64_) || (defined(_TARGET_ARM_) && !defined(LEGACY_BACKEND))
                     if ((howToPassStruct == SPK_PrimitiveType) && // Passed in a single register
```

Comment:
Doesn't this change the ABI for ARM32?

(btw, please update the `#endif` comment when updating the `#if` condition.)